source/common/common/utility.h

```diff
@@ -383,6 +383,14 @@ class StringUtil {
    */
   static std::string removeCharacters(const absl::string_view& str,
                                       const IntervalSet<size_t>& remove_characters);
+
+  /**
+   * Read a leading positive integer value. If any digits were found, store the
```

Comment:
integer -> decimal integer

Reply:
What do you think of this version?
If str begins with one or more digits, interpret them as a decimal number, and return the result. Otherwise, return std::nullopt.

Reply:
I think it also needs to explain that it will advance the start position of the string view when it successfully reads a uint64_t. Maybe this?

```
   /**                                                                                                                                                                                                                                                    
    * Read a leading positive decimal integer value and advance "*s" past the                                                                                                                                                                             
    * digits read. If overflow occurs, or no digits exist, return                                                                                                                                                                                         
    * absl::nullopt without advancing "*s".                                                                                                                                                                                                               
    */   
```